packages/kbn-analytics/src/reporter.ts

```diff
@@ -62,36 +63,31 @@ export class Reporter {
     }
   };
 
-  private log(message: unknown) {
-    if (this.debug) {
-      // eslint-disable-next-line no-console
-      console.debug(message);
-    }
-  }
-
   public reportUiCounter = (
     appName: string,
     type: string,
     eventNames: string | string[],
     count?: number
   ) => {
     const metrics = wrapArray(eventNames).map((eventName) => {
-      this.log(`${type} Metric -> (${appName}:${eventName}):`);
+      this.logger.debug(`${type} Metric -> (${appName}:${eventName}):`);
       const report = createUiCounterMetric({ type, appName, eventName, count });
-      this.log(report);
+      this.logger.debug(JSON.stringify(report));
```

Comment:
NIT we can wrap this one under `isLevelEnabled`, to avoid `JSON.stringify()` cost when not needed.

Reply:
Good call 👍